java/ql/test/library-tests/pattern-switch/cfg/Test.java

```diff
@@ -98,6 +98,33 @@ case A(B(var x, var y), var z):
          break;
      }
 
+     switch(thing) {
+       case B(_, _):
+       case Integer _, String _, A(_, _) when thing.toString().equals("abc"):
+       case Float _:
+         break;
+       default:
+         break;
+     }
+
+     var result = switch(thing) {
+       case B(_, _):
+       case Integer _, String _, A(_, _) when thing.toString().equals("abc"):
+       case Float _:
+         yield 1;
+       default:
+         yield 2;
+     };
+
+     switch ((String)thing) {
+       case "a":
+       case String _ when ((String)thing).length() == 5:
+       case "b":
+         break;
+       default:
+         break;
+     }
+
```

Comment:
As the end of a switch statement is special, I wonder if it is worth adding tests for something like
```
switch(thing) {
    case Integer:
    case String:
}
```
and
```
switch(thing) {
    case Float:
        break;
    case Integer:
    case String:
}